### PR TITLE
Add some basic installation docs

### PR DESCRIPTION
**Pull request by comzeradd**

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+CoSS development environment can be installed using **docker**. This way we run the web app and all it's dependencies as docker containers. `Here <https://www.docker.com/whatisdocker/>`_ you can find more info about what docker is.
+
+************
+Dependencies
+************
+
+#. You need to install docker in your system. The `installation guide <https://docs.docker.com/installation/#installation>`_ covers many operating systems but for now we only support Linux and Mac OS X. *Version required*: 1.3.1 or newer.
+
+#. We are using an orchestration tool for docker called `docker-compose <https://docs.docker.com/compose/>`_ that helps us automate the procedure of initiating our docker containers required for development. Installation instructions can be found `in Compose's documentation <https://docs.docker.com/compose/install/>`_. *Version required*: 1.0.1 or newer.
```
Comment (akatsoulas) [inline docs/installation.rst]: I would go for `1.0.1 or higher`

File: docs/installation.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Running Docker on Mac
+#####################
+
+Here are some notes regarding running Docker on Mac.
```
Comment (akatsoulas) [inline docs/installation.rst]: s/regarding/for

File: docs/installation.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    $ docker-compose up -d db
+
+#. Build the app container (this will take some time)::
+
+    $ docker-compose build web
+
+#. Create the database tables and run the migrations::
+
+    $ docker-compose run web python manage.py migrate --noinput
```
Comment (akatsoulas) [inline docs/installation.rst]: `docker-compose up` will take care of everything (starting the db, running the migrations etc)

Comment (comzeradd) [reply docs/installation.rst]: True, but isn't better to create a superuser before launching the website?

Comment (akatsoulas) [reply docs/installation.rst]: My comment wasn't specific to the superuser line. It's for the whole block. Probably the createsuperuser is the only thing that needs to stay here. 

That said, I don't object to have a build section. But we should make clear to the users that they don't need to follow all the steps in order to run the project.

Comment (comzeradd) [reply docs/installation.rst]: I added a notice at the beginning of the build section.

Comment (akatsoulas) [reply docs/installation.rst]: I would suggest to move the must have steps in a different section. Eg copying the env-dist file and creating the superuser account.

**Pull request merged**